tests/unit/components/playback/controls/RadioButtonWidget.spec.js

```diff
@@ -0,0 +1,94 @@
+import { mount } from "@vue/test-utils";
+import ComponentToTest from "@/components/playback/controls/player/RadioButtonWidget.vue";
+import { RadioButtonWidget as DistComponentToTest } from "@/dist/mantarray.common";
+
+import { createLocalVue } from "@vue/test-utils";
+
+let wrapper = null;
+
+const localVue = createLocalVue();
+
+let NuxtStore;
+let store;
+
+describe("RadioButtonWidget.vue", () => {
+  beforeAll(async () => {
+    // note the store will mutate across tests, so make sure to re-create it in beforeEach
+    const storePath = `${process.env.buildDir}/store.js`;
+    NuxtStore = await import(storePath);
+  });
+
+  beforeEach(async () => {
+    store = await NuxtStore.createStore();
+  });
+
+  afterEach(() => wrapper.destroy());
+  test("When mounting from the built dist file, Then it loads successfully and the props defined radio button text is rendered", () => {
+    const propsData = {
+      radio_buttons: ["warm"],
+    };
+    wrapper = mount(DistComponentToTest, {
+      propsData,
+      store,
+      localVue,
+    });
+    const target_span = wrapper.find(".custom-control-label > span");
+
+    expect(target_span.text()).toStrictEqual("warm");
+  });
+  test("When mounted, Then it loads successfully multiple defined in props as array rendered in the sequence", () => {
+    const propsData = {
+      radio_buttons: ["warm", "cool", "blue/red", "purple/green"],
+    };
+    wrapper = mount(ComponentToTest, {
+      propsData,
+      store,
+      localVue,
+    });
+    const target_span = wrapper.findAll(".custom-control-label > span");
+
+    expect(target_span.at(0).text()).toStrictEqual("warm");
+    expect(target_span.at(1).text()).toStrictEqual("cool");
+    expect(target_span.at(2).text()).toStrictEqual("blue/red");
+    expect(target_span.at(3).text()).toStrictEqual("purple/green");
+  });
+  test("Given that the radio buttons are rendered in a sequence, When a click-select of radio button occurs, Then an event 'radio-btn-selected' with index and name of the radio button is emitted", async () => {
+    const propsData = {
+      radio_buttons: ["warm", "cold"],
+    };
+    wrapper = mount(ComponentToTest, {
+      propsData,
+      store,
+      localVue,
+    });
+    const target_radio_btn = wrapper.findAll("input[type='radio']");
+    target_radio_btn.at(1).setChecked(true);
```

Comment:
is there a way to actually do a `trigger("click")` instead of `setChecked` and `trigger('change')`?

Reply:
Input[type='radio'] is using event 'change' as per the chrome V8 engine, we have to use the same.